prov/rxm/src/rxm_cq.c

```diff
@@ -134,6 +135,13 @@ static int rxm_finish_send_nobuf(struct rxm_tx_entry *tx_entry)
 		}
 		rxm_cq_log_comp(tx_entry->comp_flags);
 	}
+	
+	if (tx_entry->comp_flags & FI_SEND)
+		rxm_cntr_inc(tx_entry->ep->util_ep.tx_cntr);
+	else if (tx_entry->comp_flags & FI_WRITE)
+		rxm_cntr_inc(tx_entry->ep->util_ep.wr_cntr);
+	else
+		rxm_cntr_inc(tx_entry->ep->util_ep.rd_cntr);
```

Comment:
I'm trying to think of ways to avoid this sort of control flow except when it's needed.  util_ep has a flags field.  Maybe we can define a flag, e.g. OFI_CNTR_ENABLED, that only gets set if the EP is bound with any counter?  That would allow for checks like:
```
if (ep->util_ep.flags & OFI_CNTR_ENABLED) {
    ...
}
```

Reply:
I agree with this suggestion.
I'll do this tomorrow, if a new idea doesn't come to you :)